.github/workflows/ci.yml

```diff
@@ -26,80 +27,158 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        toxenv: [check-style, check-security, check-dependencies, check-build]
-        python-version: ['3.x']
-        os: [ubuntu-latest]
+        toxenv: [ check-style, check-security, check-dependencies, check-build ]
+        python-version: [ '3.x' ]
+        os: [ ubuntu-latest ]
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+      - uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
           cache: 'pip'
           cache-dependency-path: setup.cfg
-      - name: Run ${{ matrix.toxenv }}
-        run: |
-          pip install tox
-          tox -e ${{ matrix.toxenv }}
-
+      - run: pip install tox
+      - run: tox -e ${{ matrix.toxenv }}
   test:
-    name: ${{ matrix.toxenv }}
+    name: ${{ matrix.toxenv }} (Python ${{ matrix.python-version }}, ${{ matrix.os }})
+    needs: [ check ]
```

Comment:
You are correct; the `verify_install_requires` script only takes a few seconds, with building the environment taking up the majority of the time. Running the `toxenv` on my local machine runs within a minute. My reasoning for coupling the jobs was to prevent testing from running if a style or dependency issue would fail it anyway, but I agree that it is not worth the extra time lost. I'll work on a new PR